### ISSUE TITLE
Remove lock from documents that fail to close

### ISSUE THREAD

**Issue by nezuo** [nezuo/lapis#46]
Example implementation:
```lua
UpdateAsync(function(oldData)
    if oldData.lockId == document.lockId then
        oldData.lockId = nil
        return oldData
    end

    return nil
end)
```
As long as the current lock is still the same as the documents, it should be safe to remove it.

**Comment by nezuo** [comment]
These are the cases a document will fail to close that we should attempt to remove the session lock:
* `validate` throws an error.
* data fails to pass `validate`
* `beforeSave` errors
* `beforeClose` errors

**Comment by nezuo** [comment]
Closed by #66.
